ambry-cloud/src/main/java/com/github/ambry/cloud/CloudStorageCompactor.java

```diff
@@ -65,6 +73,29 @@ public void run() {
     }
   }
 
+  /**
+   * Shut down the compactor waiting for in progress operations to complete.
+   */
+  public void shutdown() {
+    shuttingDown.set(true);
+    logger.info("Compactor received shutdown request, waiting up to {} seconds for in-flight operations to finish",
+        shutDownTimeoutSecs);
+    try {
+      doneLatch.get().await(shutDownTimeoutSecs, TimeUnit.SECONDS);
+      logger.info("Compactor shut down successfully.");
+    } catch (InterruptedException ex) {
+      logger.warn("Timed out waiting for operations to finish.  If cloud provider uses separate stores for "
```

Comment:
Can we log a metric here? Will be easier to track. Otherwise this will get lost in log messages.